src/Connection/Connection.php

```diff
@@ -107,6 +107,9 @@ public function createPipeline($query = null, $parameters = [], $tag = null)
     public function run($statement, $parameters = null, $tag = null)
     {
         $this->checkSession();
+        if (empty($statement)) {
+            throw new \InvalidArgumentException('You must provide a valid $statement parameter');
```

Comment:
The exception message could be more clear. What do you think about changing it to

> You cannot run an empty statement.